src/tools/FreeDrawTool.js

```diff
@@ -0,0 +1,91 @@
+import Tool from "./Tool";
+
+class FreeDrawTool extends Tool {
+  constructor(model) {
+    super(model);
+    this.toolName = 'FREEDRAW';
+  }
+
+  reset() {
+    super.reset();
+    this.model.annotations.pop();
+  }
+
+  onPress( action ) {
+    // Do not propagate if control buttons are inactive. 
+    if (!this.model.controlsactive) {
```

Comment:
these two checks -- controlsactive and activityinprogress still have different code patterns in different tools. Try to unify either return or have an if then else around it that spans the full function